subsys/spm/CMakeLists.txt

```diff
@@ -13,4 +14,16 @@ if(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
   share(
     "list(APPEND ${IMAGE_NAME}BUILD_BYPRODUCTS
     ${CMAKE_BINARY_DIR}/${CONFIG_ARM_ENTRY_VENEERS_LIB_NAME})")
+    share("set(${IMAGE_NAME}CONFIG_FPU ${CONFIG_FPU})")
+endif()
+elseif(CONFIG_SPM)
+  include(${CMAKE_BINARY_DIR}/spm/shared_vars.cmake)
+  if(NOT "${CONFIG_FPU}" STREQUAL "${spm_CONFIG_FPU}")
+    add_custom_target(spm_fail
+                      COMMAND ${PYTHON_EXECUTABLE}
+                      -c \"print('Mismatch between fpu configurations in application and spm, CONFIG_FPU=${CONFIG_FPU}, spm_CONFIG_FPU=${spm_CONFIG_FPU}')\; exit(1)\"
+                      )
```

Comment:
Idea: Could this be function `BUILD_ERROR()`. Don't need to hold up the PR for this though.